tsdb/tsm1/file_store.go

```diff
@@ -1470,17 +1474,22 @@ func (c *KeyCursor) nextDescending() {
 
 	// Append the first matching block
 	if len(c.current) == 0 {
-		c.current = make([]*location, 1)
+		c.current = append(c.current, nil)
```

Comment:
reuse `c.current`, reduce slice alloc.

Reply:
This looks good @e-dard and is consistent with slice reuse in `nextAscending` per [this diff](https://github.com/influxdata/influxdb/pull/13787/files/fbd6e9f5c4506df3988ffd8b96740431c4824584#diff-48c838fa6e69378951f24571b0493a5dR1445)